File: erlang/euler20.erl
-module(euler20).
-export[euler20/0].

euler20() ->
    lists:foldl(
        fun(X, Sum) -> X + Sum end,
        0,
    	integer_to_list(fact(100))).

fact(1) -> 1;
fact(N) -> N * fact(N-1).

File: clojure/euler20.clj
; Sum of all digits in 100!

(defn fact
    ([n] (reduce *' (range 1 (inc n)))))

(defn sumdigits
   ([n] (reduce
       +'
       (map (fn [x] (read-string (str x))) (str n))))) 

(println (sumdigits (fact 100)))

